### PR TITLE
created TextureLoader and RenderMaterialLoader

### PR DESCRIPTION
**Pull request by Waterpicker**
Waterpicker Waterpicker@rocketmail.com

### CODE REVIEW

**Comment by Waterpicker** [comment]
If you think it won't work as is let me know, this is just so we something to work with.

File: engine/src/main/java/org/spout/engine/filesystem/resource/loader/RenderMaterialLoader.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
+/*
```
Comment (kitskub) [inline engine/src/main/java/org/spout/engine/filesystem/resource/loader/RenderMaterialLoader.java:1]: ...

File: engine/src/main/java/org/spout/engine/filesystem/resource/loader/RenderMaterialLoader.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+import org.spout.math.vector.Vector4;
+import org.spout.renderer.Material;
+import org.spout.renderer.data.Color;
+import org.spout.renderer.gl.Program;
+import org.spout.renderer.gl.Texture;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ *
+ * @author Johnson
```
Comment (kitskub) [inline engine/src/main/java/org/spout/engine/filesystem/resource/loader/RenderMaterialLoader.java:53]: ...

**Comment by DDoS** [comment]
This won't work. You can't create the GL objects during loading (the GL context won't always be created). The loader shouldn't even be loading GL objects from Caustic (and yes, I know the current shader loader does this, I have to fix that), we need wrappers for the API anyways.

**Pull request closed without merging**